### PR TITLE
Refs #27910 -- Improved documentation for model field choice enumeration types.

### PR DESCRIPTION
**Pull request by pope1ni**
I noticed a few typos and grammatical errors and a broken reference.

While I was there, I improved some explanations, added some helpful links and gave the enumeration types their own subsection, rather than mixing it in with rest.

### CODE REVIEW

File: docs/ref/models/fields.txt [hunk 1]
```diff
@@ -156,35 +209,44 @@ choices in a concise way::
...
 
-* Instead of values in the ``enum``, Django uses ``(value, label)`` tuples. The
-  ``label`` can be a lazy translatable string. If a tuple is not provided, the
-  label is automatically generated from the member name.
-* ``.label`` property is added on values, to return the label specified.
-* Number of custom properties are added to the enumeration classes --
+* Enum member values are a tuple of arguments to use when constructing the
+  concrete data type. Django supports adding an extra string value to the end
+  of this tuple to be used as the human-readable name, or ``label``. This
+  ``label`` can also be a lazy translatable string. Thus, in most cases, the
```
Comment (felixxm) [inline docs/ref/models/fields.txt]: I would leave the previous sentence:
```
The ``label`` can be a lazy translatable string.
```

File: docs/ref/models/fields.txt [hunk 2]
```diff
@@ -156,35 +209,44 @@ choices in a concise way::
...
+  member value will be a ``value, label`` tuple. See below for :ref:`an example
```
Comment (felixxm) [inline docs/ref/models/fields.txt]: I would leave parentheses:
```
... will be a ``(value, label)`` two-tuple.
```

File: docs/releases/3.0.txt [hunk 3]
```diff
@@ -90,7 +90,7 @@ and ``IntegerChoices`` types are provided for text and integer fields. The
 ``Choices`` class allows defining a compatible enumeration for other concrete
 data types. These custom enumeration types support human-readable labels that
 can be translated and accessed via a property on the enumeration or its
-members. See :ref:`Field.choices documentation <field-choices>` for more
+members. See :ref:`enumeration types <field-choices-enum-types>` for more
```
Comment (felixxm) [inline docs/releases/3.0.txt]: `enumeration` -> `Enumeration`

File: docs/ref/models/fields.txt [hunk 4]
```diff
@@ -219,52 +283,14 @@ you can subclass ``Choices`` and the required concrete data type, e.g.
...
-
-Unless :attr:`blank=False<Field.blank>` is set on the field along with a
-:attr:`~Field.default` then a label containing ``"---------"`` will be rendered
-with the select box. To override this behavior, add a tuple to ``choices``
-containing ``None``; e.g. ``(None, 'Your String For Display')``.
-Alternatively, you can use an empty string instead of ``None`` where this makes
-sense - such as on a :class:`~django.db.models.CharField`. To change the label
-when using one of the custom enumeration types, set the ``__empty__`` attribute
-on the class::
+- These enumeration types do not support :ref:`named groups
```
Comment (felixxm) [inline docs/ref/models/fields.txt]: I think we can chop `These`.

**Comment by pope1ni** [comment]
Updated. Thanks @felixxm.

> I think we can squash these changes into a single commit

:+1: Done. (I did it as separate commits to allow dropping bits if they weren't desired.)

**Comment by felixxm** [comment]
@pope1ni Thanks :+1: Separate commits made my review much easier, thanks for them.

**Pull request merged**